project/schema.go

```diff
@@ -0,0 +1,286 @@
+package project
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/docker/docker/pkg/nat"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+var fieldsSchemaLoader gojsonschema.JSONLoader
+var serviceSchemaLoader gojsonschema.JSONLoader
```

Comment:
I hadn't actually considered this. I'll call it in `init()`.

There's an [issue](https://github.com/xeipuuv/gojsonschema/issues/77) about thread safety, but it hasn't received any response yet. The issue is about something different, but it's actually better to cache that than a JSONLoader.